### ISSUE TITLE
[BUG] Use npm package of test library 

### ISSUE THREAD

**Issue by kavilla** [opensearch-project/opensearch-dashboards-functional-test#222]
https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package.json#L39

Utilizes the main branch when we https://www.npmjs.com/package/@opensearch-dashboards-test/opensearch-dashboards-test-library.

This causes an issue if the npm package main branch is bumped but then this repo was not expecting any change. For example the integrity of the lock file: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package-lock.json#L226

**Comment by ohltyler** [comment]
All of the github actions for plugins are currently failing due to permissions issues due to the version mismatch when the test library version was bumped. Errors seen:
```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/opensearch-project/opensearch-dashboards-test-library.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '140.82.112.4' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128
```

[ref](https://github.com/opensearch-project/opensearch-dashboards-functional-test/runs/6427808781?check_suite_focus=true)

**Comment by CCongWang** [comment]
Yeah publishing to NPM would prevent the above issues from happening, and provide another 2 benefit:

- Remove the GitHub dependency `"@opensearch-dashboards-test/opensearch-dashboards-test-library": "git+https://github.com/opensearch-project/opensearch-dashboards-test-library.git#main"`
- Have version control, which support more feature development in the future

**Comment by tianleh** [comment]
> Yeah publishing to NPM would prevent the above issues from happening, and provide another 2 benefit:
> 
>     * Remove the GitHub dependency `"@opensearch-dashboards-test/opensearch-dashboards-test-library": "git+https://github.com/opensearch-project/opensearch-dashboards-test-library.git#main"`
> 
>     * Have version control, which support more feature development in the future

Looks good! We do not need to worry about regression once we have versions for the test library.

@CCongWang Could you check with @ananzh about how the org publishes to npm? 

cc @seraphjiang @kavilla @ohltyler

**Comment by CCongWang** [comment]
@tianleh yeah we need to prepare the MCM and work with infra team to review it, will do
